tradeexecutor/analysis/trade_analyser.py

```diff
@@ -140,6 +147,25 @@ class TradePosition:
     #: but we cache it by hand to speed up processing
     closed_at: Optional[pd.Timestamp] = None
 
+    #: The total dollar value of the portfolio when the position
+    #: was opened
+    portfolio_value_at_open: Optional[USDollarAmount] = None
+
+    #: What is the maximum risk of this position.
+    #: Risk relative to the portfolio size.
+    loss_risk_at_open_pct: Optional[float] = None
+    
+    #: How much portfolio capital was risk when this position was opened.
+    #: This is based on the opening values, any position adjustment after open is ignored
+    #: Assume capital is tied to the position and we can never release it.
+    #: Assume no stop loss is used, or it cannot be trigged
+    capital_tied_at_open_pct: Optional[float] = None
```

Comment:
Maybe

- Use `portfolio_capital_at_open: Optional[USDollarAmount]` 
- `get_loss_risk_at_open_pct()`
- `get_capital_tied_at_open_pct()`

Reply:
`TradingPosition` class in state/postion.py uses `portfolio_value_at_open`, should it be changed there as well?

Reply:
I used the `get_loss_risk_at_open_pct()` and `get_capital_tied_at_open_pct()` from the `TradingPosition` class in state/position.py to directly get the percentage values for this `TradePosition` class. 

https://github.com/tradingstrategy-ai/trade-executor/blob/6d803ac9e37d2bc21f7696e27c023552e9238d40/tradeexecutor/analysis/trade_analyser.py#L1230-L1235

Should I be duplicating these methods in TradePosition as well?

Reply:
ah gotcha

Reply:
have added `maximum_risk` attirbute (equal to `position.get_loss_risk_at_open()`)